lib/galaxy/tools/parameters/grouping.py

```diff
@@ -220,7 +220,11 @@ def get_file_base_name(self, context):
         return fd
 
     def get_file_type(self, context):
-        return context.get(self.file_type_name, self.default_file_type)
+        override_file_type = context.get("override_file_type", None)
+        if override_file_type:
+            return override_file_type
+        else:
+            return context.get(self.file_type_name, self.default_file_type)
```

Comment:
Very minor but could we do something like:
`
default_type = context.get(self.file_type_name, self.default_file_type)
return context.get("override_file_type", default_type)
`

I also wonder if we could just name it `file_type` instead of `override_file_type` since they can be distinguished by being specified on request vs file bunch level.

Reply:
Yeah - so I had that at first but the fact that a value - even though it was empty - was in the parent child context means the parent context was ignored. I should probably take another swing at this though - because you are right - the way it is now kinda sucks.